### PR TITLE
Add DNSSEC checks

### PR DESCRIPTION
**Pull request by jsf9k**

### CODE REVIEW

**Comment by jsf9k** [comment]
@echudow, please give me permission to write to the `develop` branch on [echudow/trustymail](https://github.com/echudow/trustymail) and I'll merge in the latest from upstream and make the CI tests pass.

@IanLee1521 may also need write permission too.  He is a frequent contributor to [cisagov/pshtt](https://github.com/cisagov/pshtt), [cisagov/trustymail](https://github.com/cisagov/trustymail), and [18F/domain-scan](https://github.com/18F/domain-scan).

**Comment by echudow** [comment]
@jsf9k, I added you and @IanLee1521. Thanks for the help.

**Comment by jsf9k** [comment]
@echudow, I'm still not allowed to push to `develop`:
```
$ git push
Enumerating objects: 28, done.
Counting objects: 100% (28/28), done.
Delta compression using up to 8 threads
Compressing objects: 100% (16/16), done.
Writing objects: 100% (16/16), 4.36 KiB | 1.45 MiB/s, done.
Total 16 (delta 8), reused 0 (delta 0)
remote: Resolving deltas: 100% (8/8), completed with 3 local objects.
To github.com:echudow/trustymail.git
 ! [remote rejected] develop -> develop (permission denied)
error: failed to push some refs to 'git@github.com:echudow/trustymail.git'
```

**Comment by echudow** [comment]
@jsf9k, I added you as a collaborator on [echudow/trustymail](https://github.com/echudow/trustymail) and it says that it invited you, but it says that it's waiting for you to accept the invite.  You should be able to go to https://github.com/echudow/trustymail/invitations to accept the invite.

I also invited you to a private repo for building custom trust stores for use with pshtt and sslyze if you are ever interested in taking a look.  It's focused on DoD, but the extra intermediate certs could be somewhat useful.

**Comment by jsf9k** [comment]
Thanks, @echudow.  The email was in my person email account and somehow I missed it.  I was able to push just now.  I'll also take a look at the other repo.

File: trustymail/domain.py
Comment (felddy) [inline trustymail/domain.py]: This code change does not modify the logic.  It will not be executed as a list of length `0` will already evaluate to `false` and short circuit the `or`.

Comment (h-m-f-t) [reply trustymail/domain.py]: Best meta joke about DNSSEC I've seen today

:wave: :eyes: @echudow

Comment (felddy) [reply trustymail/domain.py]: 🤣

**Comment by jsf9k** [comment]
Based on the results from a full run with the new code, I think this PR is ready to be merged. I can't approve the PR myself, though, since I created it. @echudow or @IanLee1521, can either of you approve this PR?  After that we should be able to merge.

File: trustymail/domain.py
Comment (jsf9k) [reply trustymail/domain.py]: Reverted in 135833d.

Comment (echudow) [reply trustymail/domain.py]: Sorry, I'm too used to other languages where that check would have been useful.  
@h-m-f-t, I'm sure you'll find some other jokes about DNSSEC to throw my way too.

**Review by IanLee1521 — APPROVED** [review]

**Pull request merged**
